community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java

```diff
@@ -639,7 +640,11 @@ else if ( type == Double.class )
 
     private static boolean hasSafeType( Object value )
     {
-        if ( value instanceof String )
+        if ( value instanceof Value )
```

Comment:
I wonder why this whole if/else is not phrased as one single boolean expression.